Automattic/newspack-theme
# Author Bio: Display social icons from Yoast

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds the ability to show icons for social media links added to an author's profile.

Note: these fields are only available for WordPress users (and not Co-Authors Plus Guest Authors). 

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`.
2. Switch the theme to Newspack Default, and disable the Co-Authors Plus plugin if enabled.
3. Enable the Yoast plugin if not enabled -- this plugin is what provides the fields on the profile pages. 
4. Edit an author, and add a URL for each social media field (except for Twitter; add the username with no `@` symbol):

![image](https://user-images.githubusercontent.com/177561/83312494-5384d700-a1c7-11ea-8301-f4595d31485e.png)

5. Navigate to Customizer > Author Bio and check "Display Author Social Media links" and "Display Author Bio" if not already checked:

![image](https://user-images.githubusercontent.com/177561/83314550-cbef9600-a1cf-11ea-89df-e0b097b7ecc3.png)

6. Open a single post written by your author in a new tab (keeping your Customizer tab open for easy tweaks), and confirm that the icons are displayed, and they link to the links you added to the profile.

![image](https://user-images.githubusercontent.com/177561/83313752-5930eb80-a1cc-11ea-82ae-681d78bd9717.png)

7. Marvel that in this, the year 2020, you are reviewing a PR that includes a MySpace icon. 
8. Switch to the Customizer tab and check 'Display Author Email'; confirm it displays nicely next to the social icons, with a bar separating them:

![image](https://user-images.githubusercontent.com/177561/83313736-4dddc000-a1cc-11ea-9655-d318eb396277.png)

9. View the author archive page, and confirm you have the email/social links combo there, too:

![image](https://user-images.githubusercontent.com/177561/83313790-82517c00-a1cc-11ea-86a2-bde10493c72a.png)

10. Switch to the Customizer tab and uncheck 'Display Author Email' again:

![image](https://user-images.githubusercontent.com/177561/83313814-9dbc8700-a1cc-11ea-95e9-8e3dd8adaf53.png)

11. Navigate to Dashboard > Plugins, and enable Co-Authors Plus; the markup is a bit different when the plugin is enabled.
12. Repeat steps 5-10, to retest the author and confirm the social icons are displaying correctly. Note that they are not available for Guest Authors, only regular WordPress authors. 
13. Take a short break to satisfy your curiosity, and [google MySpace Tom](https://www.google.com/search?source=hp&ei=H57RXry3DZPn-gS1iZpY&q=MySpace+Tom&oq=MySpace+Tom&gs_lcp=CgZwc3ktYWIQAzIFCAAQgwEyAggAMgIIADICCAAyAggAMgIIADICCAAyAggAMgIIADICCAA6BAgAEApQwwRYghVgiBdoAXAAeACAAWaIAZ8GkgEEMTEuMZgBAKABAaoBB2d3cy13aXo&sclient=psy-ab&ved=0ahUKEwj8wv-XoNrpAhWTs54KHbWEBgsQ4dUDCAg&uact=5) to see what he's up to now.
14. Switch to Newspack Sacha. 
15. Repeat steps 5-10 with the Co-Authors Plus plugin enabled, and with it disabled:

![image](https://user-images.githubusercontent.com/177561/83314502-98ad0700-a1cf-11ea-9730-b55b4f7a8c78.png)

![image](https://user-images.githubusercontent.com/177561/83314523-ad899a80-a1cf-11ea-862e-381730eab32a.png)

![image](https://user-images.githubusercontent.com/177561/83314495-8cc14500-a1cf-11ea-943d-5bd65e81ab1f.png)


![image](https://user-images.githubusercontent.com/177561/83314432-33f1ac80-a1cf-11ea-9159-e6627065a8b5.png)




### Other information:
